icu4c/source/test/cintltst/udatpg_test.c

```diff
@@ -836,4 +840,46 @@ static void TestRegionOverride(void) {
         udatpg_close(dtpg);
     }
 }
+
+static void TestISO8601(void) {
+    typedef struct TestCase {
+        const char* locale;
+        const UChar* skeleton;
+        const UChar* expectedPattern;
+    } TestCase;
+
+    const TestCase testCases[] = {
+        { "en_GB@calendar=iso8601;rg=uszzzz", u"EEEEyMMMMdjmm", u"y MMMM d, EEEE 'at' h:mm a" },
```

Comment:
@macchiati Is it right that `j` gets us 12-hour time here?  It seems like with ISO 8601, the default hour cycle should always be 24-hour.